.github/scripts/approve-rule.mjs

```diff
@@ -22,11 +24,25 @@ if (!argv['skip-clone']) {
 }
 
 await createOrCheckoutBranch(config, argv.branch);
+await archiveRule(config, argv.ruleId);
 await generateApprovedRulePages(config, argv.ruleId);
-await updateRuleVersionsYaml(config, argv.ruleId);
+await updateRuleVersionsYaml(config, argv.ruleId, argv.changes);
 await approveTestCaseJson(config, argv.ruleId);
 await commitAndPush(config, `Set ${argv.ruleId} to approved`);
 
+async function archiveRule({ tmpDir }, ruleId) {
+  const ruleDir = `${tmpDir}content/rules/${ruleId}`;
+  if (!fs.existsSync(`${ruleDir}/index.md`)) {
+    return;
+  }
+  const ruleText = fs.readFileSync(`${ruleDir}/index.md`, 'utf8');
+  const lastModified = ruleText.match(/last_modified:\s+(.*)/)?.[1]
+  assert(lastModified, `Unable to find last_modified data in ${ruleId}/index.md`);
+
+  const date = moment(lastModified, w3cDataFormat).format(isoDateFormat)
+  await $`mv ${ruleDir}/index.md ${ruleDir}/${date}.md`
```

Comment:
TODO: This needs to be modified too: permalink, ref, & github.path need to be updated.